chainer/functions/array/hstack.py

```diff
@@ -1,5 +1,51 @@
-from chainer.functions.array import concat
-from chainer.functions.array import expand_dims
+import numpy
+
+from chainer import cuda
+from chainer import function
+from chainer.utils import type_check
+
+
+class Hstack(function.Function):
+
+    """Concatenate multiple tensors horizontally (column wise)."""
+
+    def check_type_forward(self, in_types):
+        type_check.expect(in_types.size() > 0)
+
+        ndim = in_types[0].ndim.eval()
+        for i in range(1, in_types.size().eval()):
+            type_check.expect(
+                in_types[0].dtype == in_types[i].dtype,
+                in_types[0].ndim == in_types[i].ndim,
+            )
+            if ndim <= 1:
+                continue
+            for d in range(0, ndim):
+                if d == 1:
+                    continue
+                type_check.expect(in_types[0].shape[d] == in_types[i].shape[d])
+
+    def forward(self, xs):
+        xp = cuda.get_array_module(*xs)
+        return xp.hstack(xs),
+
+    def backward(self, xs, gy):
+        if not xs[:-1]:
```

Comment:
> Certainly, it will improve the performance of the function. But I think it is more important for Chainer to implement another stacking function because Chainer has low coverage of primitive functions.

OK!
I'd give priority to implementation of `vstack` and `dstack`.

> Does it check if the length of xs is 1 or not? If so, len(xs) == 1 is easier to understand.
I did it.

In honest, I quoted it from `function.concat`.
My interpretation of `if not xs[:-1]:` is the same as yours.
So I changed it as you suggested.